liblwgeom/lwgeom_geos.c

```diff
@@ -495,92 +535,59 @@ lwgeom_geos_version()
 	return ver;
 }
 
-inline static int32_t
-get_result_srid(const LWGEOM* geom1, const LWGEOM* geom2, const char* funcname)
-{
-	if (!geom1)
-	{
-		lwerror("%s: First argument is null pointer", funcname);
-		return SRID_INVALID;
-	}
-	if (geom2 && (geom1->srid != geom2->srid))
-	{
-		lwerror("%s: Operation on mixed SRID geometries (%d != %d)", funcname, geom1->srid, geom2->srid);
-		return SRID_INVALID;
-	}
-	if (geom1->srid > SRID_MAXIMUM)
-	{
-		lwerror("%s: SRID is more than maximum (%d > %d)", funcname, geom1->srid, SRID_USER_MAXIMUM);
-		return SRID_INVALID;
-	}
-	return geom1->srid;
-}
-
-/* Input decoder and sanity checker for GEOS wrappers */
-inline static uint8_t
-input_lwgeom_to_geos(GEOSGeometry** g, const LWGEOM* geom, const char* funcname)
+/* Return the consistent SRID of all inputs, or issue an
+ * error in case of SRID mismatch. Intended to be called
+ * from RESULT_SRID macro */
+static int32_t
+get_result_srid(int count, const char* funcname, ...)
```

Comment:
Since `sizeof` returns size_t I'd rather have both `count` and `i` as size_t too.